R/g.part5.definedays.R

```diff
@@ -29,64 +31,36 @@ g.part5.definedays = function(nightsi, wi, indjump, nightsi_bu,
-    if (nightsi[1] == 1) {
-      wi = wi + 1
-      # add extra nightsi to get the last day processed (as wi has been increased by 1)
-      nightsi = c(nightsi, nightsi[length(nightsi)] + (24*(60/epochSize) * 60))
-    }
+    # if (nightsi[1] == 1) {
+    #   wi = wi + 1
+    #   # add extra nightsi to get the last day processed (as wi has been increased by 1)
+    #   nightsi = c(nightsi, nightsi[length(nightsi)] + (24*(60/epochSize) * 60))
+    # }
     NepochPerDay = ((24*3600) / epochSize)
-    if (length(nightsi) >= wi) {
-      if (wi == 1) {
-        qqq[2] = nightsi[wi] - 1
-        if (qqq[2] > NepochPerDay)  {
-          qqq[1] = qqq[2] - NepochPerDay + 1
-        } else {
-          qqq[1] = 1
-        }
-      } else if (wi <= length(nightsi)) {
-        qqq[1] = nightsi[wi - 1]
-        qqq[2] = nightsi[wi] - 1
-        qqq_backup = qqq
-      } else if (wi > length(nightsi)) {
-        qqq[1] = qqq_backup[2] + 1
-        if (wi - indjump <= length(nightsi)) {
-          tmp1 = which(nightsi_bu == nightsi[wi - indjump])
-          qqq[2] = nightsi_bu[tmp1 + indjump] - 1
-          indjump = indjump + 1 # in case there are multiple days beyond nightsi
-          if (is.na(qqq[2])) { # if that does not work use last midnight and add 24 hours
-            index_lastmidn = which(nightsi_bu == nightsi[wi - (indjump - 1)]) + (indjump - 1)
-            if (length(index_lastmidn) > 0) {
-              qqq[2] = nightsi_bu[index_lastmidn] + (24*(60/epochSize) * 60) - 1
-            } else {
-              qqq[2] = NA
-            }
-          }
-        } else {
-          qqq[2] = NA
-        }
-        if (is.na(qqq[2])) { # if that does not work use last midnight and add 24 hours
-          qqq[2] = qqq_backup[2] + (24*(60/epochSize) * 60) - 1
-        }
-        if (qqq[1] == qqq[2]) {
-          qqq[2] = qqq[2] + (24*(60/epochSize) * 60) - 1
-        }
-        if (is.na(qqq[2]) == TRUE | Nts < qqq[2]) {
-          qqq[2] = Nts
-        }
-      }
+    # if (length(nightsi) >= wi) {
+    # offset from prev midnight
+    t0 = format(ts$time[1], "%H:%M:%S")
+    hms = as.numeric(unlist(strsplit(t0, ":")))
+    NepochFromPrevMidnight = (hms[1]*60*60 + hms[2]*60 + hms[3]) / epochSize
+    if (dayborder == 0) {
+      NepochFromDayborder2Midnight = 0
     } else {
-      qqq = c(NA, NA)
-      if (length(qqq_backup) > 1) {
-        # If there is remaining time after previous day...
-        # but only do this if there is less than 24 hours.
-        # This is necessary if sleep is ignored for last night.
-        # In that case the last two calendar days should be ignored
-        # as no sleep onset will be available.
-        if (Nts - qqq_backup[2] < 24 * (60 / epochSize) * 60) {
-          qqq = c(qqq_backup[2] + 1, Nts)
-        }
-      }
+      NepochFromDayborder2Midnight = (24 - dayborder)*60*60 / epochSize
+    }
+    NepochFromPrevNight = NepochFromPrevMidnight + NepochFromDayborder2Midnight
```

Comment:
I am not sure this is correct. For example, when `dayborder = 2` and start time of the recording is 15:00, then we expect to get  `NepochFromPrevNight` equivalent to the number of epochs in 13 hours, because 15 - 2 = 13.
However, we do get the number of epochs that fit in 15 + (24 - 2) = 37 hours,  which then results in a negative qqq[2] value and causes the code to error.

Also it does not seem to be able to handle a dayborder that is after the start of the recording.

Instead of making a PR hereby the edits I made locally that seem to fix this for me:

**Replace:**

`NepochFromDayborder2Midnight = (24 - dayborder)*60*60 / epochSize`
by
`NepochFromDayborder2Midnight = (-dayborder*60*60) / epochSize`

**Followed by existing lines:**

```
}
NepochFromPrevNight = NepochFromPrevMidnight + NepochFromDayborder2Midnight
```

**and then add new lines:**

```
if (NepochFromPrevNight <= 0) {
   NepochFromPrevNight = NepochPerDay + NepochFromPrevNight
}
```

Reply:
Done, just one minor edir on your proposal, in the new lines: `if (NepochFromPrevNight <= 0) {` should be `if (NepochFromPrevNight < 0) {` so that if dayborder = 0 and start time is midnight, then `NepochPerDay` should not be summed to `NepochFromPrevNight` (which would be equal to 0). I have tested this already.